### DISCUSSION TITLE
I want my riverpod autoDispose provider to dispose of itself only when page is closed, not when page is opened

### DISCUSSION THREAD

**Discussion by anmol-me** [rrousselGit/riverpod#2459] [Q&A] [answered]
```
final selectedCategoryProvider = StateProvider.autoDispose((ref) => '');
```
I have updated this provider with a category name and then navigated to the page where its value is used. Without autoDispose the value is preserved on going to the page but with autoDispose, it uses the default value of the provider.

I cannot do away with autoDispose modifier because I want its value to be disposed of when the page is closed and its updated value before opening the page to be preserved. How to do it?

**Comment by AhmedLSayed9** [comment, marked as answer]
```dart
onPressed: () async {
  final sub = ref.listenManual(selectedCategoryProvider, (_, __) {});
  ref.read(selectedCategoryProvider.notifier).state = ...;
  await Navigator.push(..);
  sub.close();
},
```

**Reply by anmol-me** [reply]
What is listenManual()?

**Reply by AhmedLSayed9** [reply]
Same as `ref.listen` but to be used inside onPressed. https://pub.dev/documentation/flutter_riverpod/latest/flutter_riverpod/WidgetRef/listenManual.html

**Reply by anmol-me** [reply]
For me, listenManual comes with ref.

final sub = ref.listenManual(selectedCategoryProvider, (_, __) {});

**Reply by AhmedLSayed9** [reply]
Oh correct. I've copied it directly from my extension.
